Sources/View/Widget/ExpiryDatePickerView.swift

```diff
@@ -19,79 +19,91 @@
-            year = defaultYear
-        } else {
-            let currentDate = Date()
-            let currentCalendar = Calendar.current
-            month = currentCalendar.component(.month, from: currentDate)
-            year = currentCalendar.component(.year, from: currentDate)
+        setSelectedDateComponents(for: currentDate)
+        if let value = value, let date = expiryDateFormatter.date(from: value) {
+            setSelectedDateComponents(for: date)
         }
-
-        setupMonths()
-        setupYears(add: 10)
+        setupMonthsYears()
 
         delegate = self
         dataSource = self
 
-        // pick current month as default month for the picker and place the selected month in the center of picker
         selectRow(month - 1, inComponent: 0, animated: false)
+        selectRow(year % firstYear, inComponent: 1, animated: false)
     }
 
-    // MARK: UIPicker Delegate / Data Source
-    func numberOfComponents(in pickerView: UIPickerView) -> Int {
-        return 2
+    func setSelectedDateComponents(for date: Date) {
+        selectedDateComponents.calendar = calendar
+        selectedDateComponents.setValue(1, for: .day)
+        selectedDateComponents.setValue(calendar.component(.month, from: date), for: .month)
+        selectedDateComponents.setValue(calendar.component(.year, from: date), for: .year)
     }
 
+    // MARK: UIPicker Delegate / Data Source
+    func numberOfComponents(in pickerView: UIPickerView) -> Int { return 2 }
+
     func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
-        month = selectedRow(inComponent: 0) + 1
-        year = years[selectedRow(inComponent: 1)]
+        selectedDateComponents.setValue(
+            component == 0 ? row + 1 : row + firstYear,
+            for: component == 0 ? .month : .year)
     }
 
     func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
-        switch component {
-        case 0:
-            return months.count
-
-        case 1:
-            return years.count
-
-        default:
-            return 0
-        }
+        return component == 0 ? localizedMonths.count : localizedYears.count
     }
 
     func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
-        switch component {
-        case 0:
-            return months[row % months.count]
-
-        case 1:
-            return String(describing: years[row])
-
-        default:
-            return nil
-        }
-    }
-
-    private func setupMonths() {
-        months = DateFormatter().monthSymbols.map({ $0.capitalized })
+        return component == 0 ? localizedMonths[row] : localizedYears[row]
     }
 
-    private func setupYears(add: Int) {
-        years = Array(year...year + add)
+    private func setupMonthsYears(yearsRange: Int = 10) {
+        localizedMonths = localizedYearDateFormatter.standaloneMonthSymbols.map({ $0.capitalized })
```

Comment:
Why do we use this formatter for month as well? I know it should work but it's a little big confusing because in the method you set ```        formatter.setLocalizedDateFormatFromTemplate("y")
```

Reply:
Formatter provides localized months/days of week etc. according to the locale. Also, it converts from/to string/date values according to dateFormat/setLocalizedDateFormatFromTemplate and locale as well. So we can use it in both cases.